### PR TITLE
Label subtraction

### PR DESCRIPTION
**Pull request by christianbrodbeck**
Subtract labels from each other. Should be a straight forward extension from Label addition behavior.

### CODE REVIEW

File: mne/label.py [hunk 1]
```diff
@@ -341,6 +341,33 @@ def __add__(self, other):
...
+                                 '"%s" and "%s". Consider setting the '
+                                 'subject parameter on initialization, or '
+                                 'setting label.subject manually before '
+                                 'combining labels.' % (self.subject,
+                                                        other.subject))
+        else:
+            raise TypeError("Need: Label or BiHemiLabel. Got: %r" % other)
+
+        if self.hemi == other.hemi:
+            keep = np.in1d(self.vertices, other.vertices, True, invert=True)
```
Comment (agramfort) [inline mne/label.py]: use in1d from fixes

**Comment by dengemann** [comment]
@christianbrodbeck this closes and issue. Let's find and close it on merge.

**Comment by christianbrodbeck** [comment]
The error seems to be unrelated

```
ERROR: Test file downloading
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/miniconda/envs/testenv/lib/python2.6/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/home/travis/build/mne-tools/mne-python/mne/utils.py", line 589, in dec
    return function(*args, **kwargs)
  File "/home/travis/build/mne-tools/mne-python/mne/tests/test_utils.py", line 363, in test_fetch_file
    _fetch_file(url, archive_name, verbose=False)
  File "<string>", line 2, in _fetch_file
  File "/home/travis/build/mne-tools/mne-python/mne/utils.py", line 534, in verbose
    return function(*args, **kwargs)
  File "/home/travis/build/mne-tools/mne-python/mne/utils.py", line 1376, in _fetch_file
    verbose_bool=verbose_bool)
  File "/home/travis/build/mne-tools/mne-python/mne/utils.py", line 1296, in _chunk_read_ftp_resume
    data.retrbinary(down_cmd, chunk_write)
  File "/home/travis/miniconda/envs/testenv/lib/python2.6/ftplib.py", line 403, in retrbinary
    conn = self.transfercmd(cmd, rest)
  File "/home/travis/miniconda/envs/testenv/lib/python2.6/ftplib.py", line 365, in transfercmd
    return self.ntransfercmd(cmd, rest)[0]
  File "/home/travis/miniconda/envs/testenv/lib/python2.6/ftplib.py", line 331, in ntransfercmd
    conn = socket.create_connection((host, port), self.timeout)
  File "/home/travis/miniconda/envs/testenv/lib/python2.6/socket.py", line 561, in create_connection
    raise error, msg
error: [Errno 110] Connection timed out
```

**Comment by christianbrodbeck** [comment]
I realized that I could restart Travis, now it's passing. @agramfort 's comment is addressed.

**Comment by agramfort** [comment]
please update what's new. Then +1 for merge.

thanks @christianbrodbeck

**Comment by dengemann** [comment]
+1 for merge too.

**Comment by christianbrodbeck** [comment]
updated whats_new

**Comment by agramfort** [comment]
thanks @christianbrodbeck

**Pull request merged**